docs/sc-memory/api/cpp/extended/agents/actions.md

```diff
@@ -0,0 +1,345 @@
+action.SetArgument(ScKeynodes::rrel_1, argAddr);
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+action.SetArgument(1, argAddr);
+...
+```
+
+!!! note
+    If action has already argument with specified role, then connection between action and this argument will be removed and created new one between action and new argument.
+
+### **SetArguments**
+
+```cpp
+...
+// Set several arguments simultaneously.
+action.SetArguments(argAddr1, argAddr2);
+...
+```
+
+!!! note
+    `SetArguments` uses `SetArgument` for each argument.
+
+### **Action result**
+
+All action should have result (result situation). Result situation is structure that contains all sc-constructions that which indicate result of interpreting action.
+
+#### **GetResult**
+
+You can result for any actions.
+
+```cpp
+...
+// Use this method after that some agent finished 
+// interpreting action.
+ScStructure const & actionResult = action.GetResult();
+...
+```
+
+!!! warning
+    You can call this method for finished action only.
+
+#### **SetResult**
+
+```cpp
+...
+// Use this method in agent interpreting action 
+// to set new result.
+action.SetResult(resultStructure);
+...
+```
+
+!!! note
+    If action has result, then will be removed and the new one will be set.
+
+!!! warning
+    You can call this method for not finished, but initiated action only.
+
+#### **FormResult**
+
+You must not create result structure. You can provide only elements of result for action.
+
+```cpp
+...
+// Use this method in agent interpreting action 
+// to form new result with provided sc-elements.
+action.FormResult(elementAddr1, elementAddr2);
+...
+```
+
+!!! note
+    If action has result, then will be removed and the new one will be set.
+
+#### **UpdateResult**
+
+```cpp
+...
+// Use this method in agent interpreting action
+// for update existing result by new sc-elements.
+action.UpdateResult(elementAddr1, elementAddr2);
+...
+```
+
+!!! note
+    This method updates existing result for action.
+
+!!! note
+    `FormResult` and `UpdateResult` doesn't append sc-element twice.
+
+!!! note
+    If you don't form result then empty result for your action will be generated if you finish action.
+
+### **Action states**
+
+All actions have state. There are three states of actions provided by this API:
+
+* action isn't initiated;
+* action is initiated, but isn't finished, that is, action is performed;
+* action is finished.
+
+You can initiate, wait or finish actions.
+
+#### **IsInitiated**
+
+Use this method to check that specified action is initiated.
+
+```cpp
+...
+sc_bool const isActionInitiated = action.IsInitiated();
+...
+```
+
+#### **InitiateAndWait**
+
+You can initiate action and wait while it will be finished.
+
+```cpp
+...
+// Provide maximum time of waiting while action will be finished.
+action.InitiateAndWait(100); // milliseconds
```

Comment:
what do you think about creating methods like WaitMilliseconds(), WaitSecond() and others so it will be clear what to pass to them

Reply:
I propose to create `ScTime` class and override this method with parameter of this class. It will more clear and expandable

Reply:
I'll create issue